### PR TITLE
Lae 63 payment gateway integration

### PR DESCRIPTION
**Pull request by salniukas**

### CODE REVIEW

**Comment by salniukas** [comment]
[[Revert "tweak: renaming? not sure what happend"]](https://github.com/dovydasvg/NaturaMedica/commit/7395379c2e8b4dc04d3699134f4bce52092cc136) 


VScode decided to rename files, which caused build Issues.


Reverted this commit.

File: src/server/api/routers/checkout.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
+import { useAtom } from "jotai";
+import { z } from "zod";
+import { env } from "~/env";
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import { cartAtom } from "~/utils/Atoms";
+
+type Order = {
```
Comment (dovydasvg) [inline src/server/api/routers/checkout.ts:7]: move checkout.ts to checkout folder like checkout/index.ts and move types to types.ts within that folder. This is just too big, so let's move them out to keep this clean.

File: src/server/api/routers/checkout.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
+import { useAtom } from "jotai";
+import { z } from "zod";
+import { env } from "~/env";
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import { cartAtom } from "~/utils/Atoms";
```
Comment (dovydasvg) [inline src/server/api/routers/checkout.ts:5]: ? 🤔

File: src/server/api/routers/checkout.ts [hunk 3]
```diff
@@ -0,0 +1,180 @@
+import { useAtom } from "jotai";
```
Comment (dovydasvg) [inline src/server/api/routers/checkout.ts:1]: ? 🤣

File: src/server/api/routers/checkout.ts [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      total: z.string(),
+    }),
+  ),
+});
+
+export const CheckoutRouter = createTRPCRouter({
+  CreateOrder: publicProcedure
+    .input(OrderSchema)
+    .mutation(async ({ input }) => {
+      const CreatedOrder = await CreateOrder(input as Order);
```
Comment (dovydasvg) [inline src/server/api/routers/checkout.ts:107]: camelCase everything here. We never use Pascal casing 🙏

File: package-lock.json [hunk 5]
```diff
@@ -20,6 +20,7 @@
         "class-variance-authority": "^0.7.0",
         "clsx": "^2.1.0",
         "dayjs": "^1.11.11",
+        "embla-carousel-auto-scroll": "^8.0.4",
```
Comment (dovydasvg) [inline package-lock.json:23]: weird that you've had changes here? 🤔 pnpm i ? 🤔 This file shouldn't be pushed, as there were no changes in package.json 🙏

**Review by dovydasvg — CHANGES_REQUESTED** [review]

File: src/server/api/routers/checkout/index.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
+import { env } from "~/env";
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import type { createdOrder, order, orderJSON } from "./types";
+import { orderSchema } from "./types";
+
+export const CheckoutRouter = createTRPCRouter({
```
Comment (dovydasvg) [inline src/server/api/routers/checkout/index.ts:6]: router to camelCase

**Review by dovydasvg — CHANGES_REQUESTED** [review]

**Review by dovydasvg — APPROVED** [review]

**Pull request merged**